spec.bs

```diff
@@ -494,6 +506,253 @@ The <dfn for=Navigator method>runAdAuction(|config|)</dfn> method steps are:
+
+  : [=fenced frame config/fenced frame reporting metadata=]
+  :: null
+
+  : [=fenced frame config/exfiltration budget metadata=]
+  :: null
+
+  : [=fenced frame config/nested configs=]
+  :: a [=struct=] with the following [=struct/items=]:
+    : [=nested configs/value=]
+    :: an empty [=list=] «»
+
+    : [=nested configs/visibility=]
+    :: "<a for=visibility>`opaque`</a>"
+
+  : [=fenced frame config/embedder shared storage context=]
+  :: null
+
+</div>
+
+<div algorithm="fill in a pending fenced frame config">
+
+To <dfn>fill in a pending fenced frame config</dfn> given a [=fenced frame config=]
+|pendingConfig|, [=auction config=] |auctionConfig|, and [=leading bid info=] |winningBidInfo|:
+1. Let |winningBid| be |winningBidInfo|'s [=leading bid info/leading bid=].
+1. Set |pendingConfig|'s [=fenced frame config/mapped url=]'s [=mapped url/value=] to
+   |winningBid|'s [=generated bid/ad descriptor=]'s [=ad descriptor/url=].
+1. Let |adSize| be |winningBid|'s [=generated bid/ad descriptor=]'s [=ad descriptor/size=].
+1. If |adSize| is not null:
+   1. Set |pendingConfig|'s [=fenced frame config/content size=] to a [=struct=] with the following
+      [=struct/items=]:
+     : [=content size/value=]
+     :: |adSize| TODO: Resolve screen-relative sizes and macros and cast this properly.
+
+     : [=content size/visibility=]
+     :: "<a for=visibility>`opaque`</a>"
+1. Set |pendingConfig|'s [=fenced frame config/interest group descriptor=]'s
+   [=interest group descriptor/value=] to a [=struct=] with the following [=struct/items=]:
+     : owner
+     :: |winningBid|'s [=generated bid/interest group=]'s [=interest group/owner=]
+
+     : name
+     :: |winningBid|'s [=generated bid/interest group=]'s [=interest group/name=]
+1. Set |pendingConfig|'s [=fenced frame config/fenced frame reporting metadata=] to a [=struct=]
+   with the following [=struct/items=]:
+    : [=fenced frame reporting metadata/value=]
+    :: If |auctionConfig|'s [=auction config/component auctions=] is [=list/empty=] (i.e., if
+       there was no component auction), then a [=struct=]  with the following [=struct/items=]:
+       : [=fenced frame reporting metadata/fenced frame reporting map=]
+       :: a [=map=] «[ "<code>buyer</code>" → «», "<code>seller</code>" → «»]»
+
+       : [=fenced frame reporting metadata/direct seller is seller=]
+       :: true
+
+       Otherwise (i.e., if there was a component auction), a [=struct=] with the following
+       [=struct/items=]:
+       : [=fenced frame reporting metadata/fenced frame reporting map=]
+       :: a [=map=] «[ "<code>buyer</code>" → «», "<code>seller</code>" → «»,
+          "<code>component-seller</code>" → «»]»
+
+       : [=fenced frame reporting metadata/direct seller is seller=]
+       :: false
+
+    : [=fenced frame reporting metadata/visibility=]
+    :: "<a for=visibility>`opaque`</a>"
+1. Set |pendingConfig|'s [=fenced frame config/on navigate callback=] to a closure of

```

Comment:
What I was saying in my other thread is that we should avoid the word "closure" here since that's not really a formal Infra thing at the moment. We should instead make this:

```
Set |pendingConfig|'s [=fenced frame config/on navigate callback=] to an algorithm with these steps:
    1. [=Asynchronously finish reporting=] with |pendingConfig|'s [.....]
```